mlflow/tracing/fluent.py

```diff
@@ -260,7 +279,184 @@ def pagination_wrapper_func(number_to_get, next_page_token):
+        fields: A list of field strings of the form 'span_name.[inputs|outputs]' or
+            'span_name.[inputs|outputs].field_name'.
+        col_name: The name of the column in the traces DataFrame containing the spans. If `traces`
+            is a list of MLflow Traces, this argument should not be provided.
+    """
+    try:
+        import pandas as pd
+    except ImportError as e:
+        raise MlflowException(
+            message=(
+                "The `pandas` library is not installed. Please install `pandas` to use the"
+                f"`mlflow.tracing.extract` function. Error: {e}"
+            ),
+        )
+
+    parsed_fields = _parse_fields(fields)
+
+    if isinstance(traces, list):
+        if col_name is not None:
+            raise MlflowException(
+                message=(
+                    "If `traces` is a list of MLflow Traces, `col_name` should not be provided."
+                ),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        traces = traces_to_df(traces)
+        col_name = SPANS_COLUMN_NAME
+
+    if isinstance(traces, pd.DataFrame):
+        return _extract_from_traces_pandas_df(df=traces, col_name=col_name, fields=parsed_fields)
+
+    raise MlflowException(
+        message=(
+            "`traces` must be a list of MLflow Traces or a pandas DataFrame. Got: {type(traces)}"
+        ),
+        error_code=INVALID_PARAMETER_VALUE,
+    )
+
+
+class _ParsedField(NamedTuple):
+    """
+    Represents a parsed field from a string of the form 'span_name.[inputs|outputs]' or
+    'span_name.[inputs|outputs].field_name'.
+    """
+
+    span_name: str
+    field_type: Literal["inputs", "outputs"]
+    field_name: Optional[str]
+
+    @classmethod
+    def from_string(cls, s: str) -> "_ParsedField":
+        components = s.split(".")
+        if len(components) not in [2, 3] or components[1] not in ["inputs", "outputs"]:
+            raise MlflowException(
+                message=(
+                    f"Field must be of the form 'span_name.[inputs|outputs]' or"
+                    f" 'span_name.[inputs|outputs].field_name'. Got: {s}"
+                ),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+        return cls(
+            span_name=components[0],
+            field_type=components[1],
+            field_name=components[2] if len(components) == 3 else None,
+        )
+
+    def __str__(self) -> str:
+        return (
+            f"{self.span_name}.{self.field_type}.{self.field_name}"
+            if self.field_name is not None
+            else f"{self.span_name}.{self.field_type}"
+        )
+
+
+def _parse_fields(fields: List[str]) -> List["_ParsedField"]:
+    """
+    Parses the specified field strings of the form 'span_name.[inputs|outputs]' or
+    'span_name.[inputs|outputs].field_name' into _ParsedField objects.
+    """
+    return [_ParsedField.from_string(field) for field in fields]
+
+
+def _extract_from_traces_pandas_df(
```

Comment:
nit: Can we move these utility methods and dataclass to separate utility file, and add some unit tests? These are complex functions and seems to have edge cases. Testing them separately is easier than adding larger tests for `search_traces`.